src/o365/spo/commands/search/search.ts

```diff
@@ -0,0 +1,434 @@
+    const enablePhoneticRequestString = typeof(args.options.enablePhonetic) === 'undefined' ? `` : `&enablephonetic=${args.options.enablePhonetic}`;    
+    const processBestBetsRequestString = typeof(args.options.processBestBets) === 'undefined' ? `` : `&processbestbets=${args.options.processBestBets}`;
+    const enableQueryRulesRequestString = typeof(args.options.enableQueryRules) === 'undefined' ? `` : `&enablequeryrules=${args.options.enableQueryRules}`;
+    const processPersonalFavoritesRequestString = typeof(args.options.processPersonalFavorites) === 'undefined' ? `` : `&processpersonalfavorites=${args.options.processPersonalFavorites}`;
+
+    //Construct single requestUrl
+    const requestUrl = `${webUrl}/_api/search/query?querytext='${args.options.query}'`.concat(
+      propertySelectRequestString,
+      startRowRequestString,
+      rowLimitRequestString,
+      sourceIdRequestString,
+      trimDuplicatesRequestString,
+      enableStemmingRequestString,
+      cultureRequestString,
+      refinementFiltersRequestString,
+      queryTemplateRequestString,
+      sortListRequestString,
+      rankingModelIdRequestString,
+      propertiesRequestString,
+      refinersRequestString,
+      hiddenConstraintsRequestString,
+      clientTypeRequestString,
+      enablePhoneticRequestString,
+      processBestBetsRequestString,
+      enableQueryRulesRequestString,
+      processPersonalFavoritesRequestString
+    );
+
+    if(this.debug) {
+      cmd.log(`RequestURL: ${requestUrl}`);
+    }
+    return requestUrl;
+  }
+
+  private getPropertiesRequestString(args: CommandArgs):string {
+    let properties = args.options.properties ? args.options.properties : '';
+    if(args.options.sourceName) {
+      if(properties && !properties.endsWith(",")) { properties += `,`; }
+      properties += `SourceName:${args.options.sourceName},SourceLevel:SPSite`;
+    }
+    return properties ? `&properties='${properties}'` : ``;
+  }
+
+  private getSelectPropertiesArray(args: CommandArgs) {
+    return args.options.selectProperties 
+      ? args.options.selectProperties.split(",")
+      : (!args.options.output || args.options.output === "text") 
+        ? ["Rank","DocId","OriginalPath"/*,"PartitionId","UrlZone","Culture","ResultTypeId","RenderTemplateId"*/]
+        : [];
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-q, --query <query>',
+        description: 'Query to be executed in KQL format'
+      },
+      {
+        option: '-p, --selectProperties <selectProperties>',
+        description: 'Comma separated list of properties to retrieve. Will retrieve all properties if not specified and json output is requested.'
+      },
+      {
+        option: '--allResults',
+        description: 'Get all results of the search query, not only the amount specified by the rowlimit (default: 10)'
```

Comment:
Let's rephrase it to `Set, to get all results...`. That way it's clearer what it does.